modules/ting_field_search/ting_field_search.module

```diff
@@ -91,54 +151,14 @@ function ting_field_search_get_active_profile() {
   return $profile;
 }
 
-/**
- * Set the current search request type when a profile was activated.
- */
-function ting_field_search_set_context($request_type) {
-  $static = &drupal_static(__FUNCTION__);
-  $static = $request_type;
-}
-
-/**
- * If a profile is active this returns the associated ting request type.
- */
-function ting_field_search_get_context() {
-  return drupal_static('ting_field_search_set_context');
-}
-
-/**
- * Implements hook_opensearch_cache_key_alter().
- *
- * If we have an active profile and context, we use this hook to modify the
- * cache key, so that ting_cache can fetch/save the correct result.
- *
- * @see ting.api.php
- */
-function ting_field_search_opensearch_cache_key_alter(&$cid, $request) {
-  if (!ting_field_search_get_context()) {
-    return;
-  }
-  if ($profile = ting_field_search_get_active_profile()) {
-    $cid .= ':' . $profile->name;
-  }
-}
-
 /**
  * Implements hook_opensearch_pre_execute().
  */
 function ting_field_search_opensearch_pre_execute($request) {
-  if (!$context = ting_field_search_get_context()) {
-    return;
-  }
-
-  // Do we support the class of this request?
-  $class = get_class($request);
-  $supported_classes = array(
-    'TingClientObjectRequest',
-    'TingClientCollectionRequest',
-    'TingClientSearchRequest',
-  );
-  if (!in_array($class, $supported_classes)) {
```

Comment:
I do not think it is a good idea to move this code to the hook implementation. The code in this function has some expectations regarding what methods can be called on the request e.g. `setProfile()`. Some of these methods are not available on all types of requests. Consequently I think it should rely on this method on this function to filter away irrelevant request types instead of relying on a hook implementation to handle it.

Reply:
Done. I agree it makes more sense to have in `ting_field_opensearch_pre_execute()`.